src/main/java/org/terasology/behaviors/system/FindNearbyPlayersSystem.java

```diff
@@ -46,47 +51,48 @@
             Vector3f actorPosition = entity.getComponent(LocationComponent.class).getWorldPosition();
             FindNearbyPlayersComponent findNearbyPlayersComponent = entity.getComponent(FindNearbyPlayersComponent.class);
             float maxDistance = findNearbyPlayersComponent.searchRadius;
             float maxDistanceSquared = maxDistance * maxDistance;
-            Iterable<EntityRef> clients = entityManager.getEntitiesWith(ClientComponent.class);
-            List<EntityRef> charactersWithinRange = Lists.newArrayList();
-
-            EntityRef closestCharacter = EntityRef.NULL;
-            float minDistanceFromCharacter = 0.0f;
 
-            for (EntityRef client : clients) {
-                ClientComponent clientComponent = client.getComponent(ClientComponent.class);
-                EntityRef character = clientComponent.character;
-                AliveCharacterComponent aliveCharacterComponent = character.getComponent(AliveCharacterComponent.class);
-                if (aliveCharacterComponent == null) {
-                    continue;
-                }
-                LocationComponent locationComponent = character.getComponent(LocationComponent.class);
-                if (locationComponent == null) {
-                    continue;
-                }
-                if (locationComponent.getWorldPosition().distanceSquared(actorPosition) <= maxDistanceSquared) {
-                    if (charactersWithinRange.size() == 0) {
-                        closestCharacter = character;
-                        minDistanceFromCharacter = locationComponent.getWorldPosition().distanceSquared(actorPosition);
-                    } else {
-                        if (locationComponent.getWorldPosition().distanceSquared(actorPosition) < minDistanceFromCharacter) {
-                            closestCharacter = character;
-                            minDistanceFromCharacter = locationComponent.getWorldPosition().distanceSquared(actorPosition);
-                        }
-                    }
-                    charactersWithinRange.add(character);
-                }
+            List<Vector3f> inRange = locationSet.stream()
+                    .filter(loc -> loc.distanceSquared(actorPosition) <= maxDistanceSquared)
+                    .collect(Collectors.toList());
+            if (inRange.isEmpty()) {
+                findNearbyPlayersComponent.charactersWithinRange = Collections.emptyList();
+                findNearbyPlayersComponent.closestCharacter = EntityRef.NULL;
+                entity.saveComponent(findNearbyPlayersComponent);
+                continue;
             }
 
+            inRange.sort(Comparator.comparingDouble(v3f -> v3f.distanceSquared(actorPosition)));
+            List<EntityRef> charactersWithinRange = new ArrayList<>(inRange.size());
+            inRange.forEach(v3f -> charactersWithinRange.add(clientLocationMap.get(v3f)));
```

Comment:
Why not use Streams-API here, too? Isn't this the same as:

```suggestion
            List<EntityRef> charactersWithinRange = 
            	inRange.stream().map(clientLocationMap::get).collect(Collectors.toList());;
```